test/Test/Pos/Util/ModifierSpec.hs

```diff
@@ -0,0 +1,185 @@
+                \ whose key is not present in the modifier"
+                (allPairsAreInAssocList @Text @Text)
+        describe "insertionsMap" $ do
+            prop
+                "the created hashmap removes all the deleted keys from the 'MapModifier'\
+                \ and keeps all the inserted keys"
+                (mapModifierToHashMap @Text @Text)
+        describe "insertions" $ do
+            prop
+                "the created list removes all the deleted keys from the used\
+                \ 'MapModifier' and keeps all the inserted keys in the association list"
+                (mapModifierToList @Text @Text)
+        describe "deletions" $ do
+            prop
+                "the created list of keys removes all the inserted values and keeps those\
+                \ which have been deleted"
+                (mapModifierToDeletionList @Text @Text)
+        describe "mapMaybe" $ do
+            prop
+                "the created list of keys removes all the deleted values and keeps the\
+                \ pairs from the input list whose keys are not in the modifier"
+                (allPairsAreInMaybeList @Text @Text @Text Just)
+        describe "modifyHashMap" $ do
+            prop "modifying a hashmap with a map modifier removes the keys that were\
+                 \ deleted and keeps those which were inserted, along with their values"
+                 (hashMapIsModified @Text @Text)
+
+
+mapModifierFormsMonoid
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Property
+mapModifierFormsMonoid = formsMonoid
+
+insertThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => k
+    -> v
+    -> v
+    -> HM.HashMap k v
+    -> Core.MapModifier k v
+    -> Property
+insertThenLookup k v1 v2 hm mapMod =
+    let newMap = Core.insert k v1 mapMod
+        newHMap = HM.insert k v2 hm
+    in v1 /= v2 ==> maybe False (== v1) $ Core.lookup (flip HM.lookup newHMap) k newMap
+
+insertDeleteThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => HashMap k v
+    -> k
+    -> v
+    -> Core.MapModifier k v
+    -> Bool
+insertDeleteThenLookup m k v mapMod =
+    let newMap = HM.insert k v m
+        newMapMod = Core.delete k mapMod
+    in isNothing $ Core.lookup (flip HM.lookup newMap) k newMapMod
+
+allKeysAreInMap
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => HM.HashMap k v
+    -> Core.MapModifier k v
+    -> Bool
+allKeysAreInMap hm mapMod =
+    let keys = Core.keys (HM.keys hm) mapMod
+    in all (\k -> (isJust $ Core.lookup (flip HM.lookup hm) k mapMod)) keys
+
+allValsAreInMap
+    :: (Ord k, Show k, Show v, Eq k, Eq v, Hashable k)
+    => [(k, v)]
+    -> Core.MapModifier k v
+    -> Bool
+allValsAreInMap kvs mapMod =
+    let vals = Core.values kvs mapMod
+    in all (\v -> (v `elem` (map snd (Core.insertions mapMod))) ||
+                  (v `elem` fmap snd kvs)
```

Comment:
The `values` function isn't going to return deleted values, is it? Is that a case we need to worry about?